frontend/public/components/storage-class-form.tsx

```diff
@@ -64,105 +65,29 @@ export class StorageClassForm_ extends React.Component<
 
   storageTypes = {};
 
+  // Fetch Storage type provisoners from different operators
+  // For CSI - provionerType: 'csi'
+  // For Defaults - provionerType: 'other'
+  getExtensionsStorageClassProvisioners = (provionerType: string) => {
+    const extensionCSIProvisoners = {};
+    _.forEach(plugins.registry.getStorageClassProvisioner(), (operator: any) => {
+      if (operator) {
+        _.forEach(operator.properties.getStorageClassProvisoner(), (params, type) => {
+          extensionCSIProvisoners[type] = params;
+        });
+      }
+    });
+    return extensionCSIProvisoners[provionerType];
+  };
```

Comment:
Use the `useExtension` hook.
`const scProvisioners  = useExtension<StorageClassProvisioner>(isStorageClassProvisioner);`

Reply:
Hooks cannot be used inside class components.

Reply:
Use `withExtensions` HOC.

Reply:
I see. This is still a class based component.

Reply:
I understand `withExtensions` will work here, but the aim is to do minimal changes in public component to add the extension.